syntax = "proto3";

package tripsphere.attraction;

import "tripsphere/common/geo.proto";
import "tripsphere/file/file.proto";

option go_package = "tripsphere/attraction";
option java_multiple_files = true;
option java_package = "org.tripsphere.attraction";

message Attraction {
  string id = 1;
  string name = 2;
  tripsphere.common.Address address = 3;
  string introduction = 4;
  repeated string tags = 5;
  repeated tripsphere.file.File images = 6;
  tripsphere.common.Location location = 7;
}

message AddAttractionRequest {
  Attraction attraction = 1;
}

message AddAttractionResponse {
  string id = 1; //attraction id
}

message DeleteAttractionRequest {
  string id = 1;
}

message DeleteAttractionResponse {
  bool success = 1;
}

message ChangeAttractionRequest {
  Attraction attraction = 1;
}

message ChangeAttractionResponse {
  bool success = 1;
}

message FindIdByNameRequest {
  string name = 1;
}

message FindIdByNameResponse {
  string attraction_id = 1;
}

service AttractionService {
  rpc AddAttraction(AddAttractionRequest) returns (AddAttractionResponse);
  rpc DeleteAttraction(DeleteAttractionRequest) returns (DeleteAttractionResponse);
  rpc ChangeAttraction(ChangeAttractionRequest) returns (ChangeAttractionResponse);
  rpc FindIdByName(FindIdByNameRequest) returns (FindIdByNameResponse);
}
